### ISSUE TITLE
Implement battle demo using demo design document.

### ISSUE THREAD

**Issue by aliu65803** [uchicago-cs/chiventure#1530]
For this issue, we will implement our demo by following the design outlined in our design document. We will use the previous examples as a template for our demo. We will create one room that has two enemies, meaning that the user can fight one after the other if they so choose. In order to showcase our new battle equipment struct, one enemy will have battle equipment and one will not. We will aim to have the battle CLI example as the demo, and have the other two examples as backup plans in case the CLI demo does not work.

**Comment by aliu65803** [comment]
We are done creating the items, equipment, and moves for the demo.

**Comment by nicholaswlee** [comment]
Last night, we met up and decided to try to remake the examples from last year to include our new stuff. Currently two of the run: battle_command_example and battle_flow_example. In order to finish the CLI one, we will have to wait for run_battle to be implemented.

**Comment by nicholaswlee** [comment]
After some more debugging, our battle_command_example is working!!! We are able to consume items, throw items at our opponent, and have a successful battle that show cases stat changes and other new battle mechanics. This example will be great for the next team, and a good backup plan just in case battle_cli_example.c doesn't work.

**Comment by nicholaswlee** [comment]
We are now currently working on battle_cli_example.c. Most of our issues are with segmentation faults that we have been debugging with gdb.

**Comment by nicholaswlee** [comment]
The battle_cli_example.c is able to run a very limited functioning battle. However, it does not allow for any mistakes in input and creates a lot of segmentation faults. We are prioritizing faulty inputs and allowing for mistakes from our users now.

**Comment by nicholaswlee** [comment]
We have gotten all examples to work and a PR has been submitted. Most of the segfaults have been resolved, although some still exist and will have to be avoided during the demo.

**Comment by aliu65803** [comment]
The PR has successfully been merged, completing our work for this issue. For this issue, we have changed the previous examples to reflect the changes we have made throughout the term including updates to battle equipment, stats, items, and more. For our demo, we would like to showcase our battle CLI example, which will allow the user to input their actions through the CLI and run a battle that uses turn components. If this fails, we have a backup plan to use the other two examples to showcase our work.
